cmd/ignited/ignited.go

```diff
@@ -2,19 +2,47 @@ package main
 
 import (
 	"os"
+	"os/signal"
+	"syscall"
 
+	log "github.com/sirupsen/logrus"
 	"github.com/weaveworks/ignite/cmd/ignited/cmd"
 	"github.com/weaveworks/ignite/pkg/providers"
 	"github.com/weaveworks/ignite/pkg/providers/ignited"
 	"github.com/weaveworks/ignite/pkg/util"
 )
 
 func main() {
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sigChan
+
+		log.Debug("Signal SIGTERM caught")
+		cleanup()
+
+		log.Debug("Program normally terminated")
+		os.Exit(0)
+	}()
+
 	if err := Run(); err != nil {
+		log.Debugf("Termination with error: %v", err)
 		os.Exit(1)
 	}
 }
 
+func cleanup() {
+
+	const daemonSock = "/var/lib/firecracker/daemon.sock"
```

Comment:
The reconcile file uses constants from constants.DATA_DIR, constants.DAEMON_SOCKET.  we should do that instead of hard coding it here.

Reply:
Fixed. Thank you.